Modules/_curses_panel.c

```diff
@@ -648,14 +660,19 @@ static PyMethodDef PyCursesPanel_Methods[] = {
 
 static PyType_Slot PyCursesPanel_Type_slots[] = {
     {Py_tp_dealloc, PyCursesPanel_Dealloc},
+    {Py_tp_traverse, PyCursesPanel_Traverse},
     {Py_tp_methods, PyCursesPanel_Methods},
     {0, 0},
 };
 
 static PyType_Spec PyCursesPanel_Type_spec = {
     .name = "_curses_panel.panel",
     .basicsize = sizeof(PyCursesPanelObject),
-    .flags = Py_TPFLAGS_DEFAULT | Py_TPFLAGS_DISALLOW_INSTANTIATION,
+    .flags = (
+        Py_TPFLAGS_DEFAULT
+        | Py_TPFLAGS_DISALLOW_INSTANTIATION
```

Comment:
We'll also add the immutable flag.

Reply:
You can create a reference loop using `panel.set_userptr()`. To break such loops we need a `tp_clear` that sets a userptr to NULL (see `PyCursesPanel_Dealloc()`). It should not clear `self->wo`! `tp_traverse` also should call `panel_userptr()` and visit its result.

It is better to make many types immutable in a separate PR if we are not going to backport that change.

Reply:
I see, thank you!